web/src/games/cardtable/game.ts

```diff
@@ -0,0 +1,194 @@
+import { Ctx } from 'boardgame.io';
+import { INVALID_MOVE } from 'boardgame.io/core';
+import { dealCribbage } from './deals';
+
+export interface IG {
+  count: number;
+  deck: ICard[];
+  hands: ITable;
+  stock: ICard[];
+}
+
+export interface ITable {
+  north: IHand;
+  south: IHand;
+  east: IHand;
+  west: IHand;
+}
+
+export interface ICard {
+  id: string;
+  rank: number;
+  faced: boolean;
+  img: string;
+}
+
+export interface IHand {
+  private: ICard[];
+  held: ICard[];
+  melds: ICard[];
+  tricks: ICard[];
+  played: ICard[];
+  cribFlipped?: boolean;
```

Comment:
what is a crib?

Reply:
The act of 'laying-to-crib' is a reference to that relationship and laying your cards into a place where they're kept in secret during cribbage play; the dealer and the opponent(s) both lay to "the crib" secretly and the dealer is given an opportunity to score on their value as the final phase of hand play.

Cribbage takes it name from the practice of laying up ships/boats in cribbing when in dry dock.  A crib is also a childs bed shaped like those dockside affairs \__/